gammapy/utils/interpolation.py

```diff
@@ -92,9 +94,21 @@ def __call__(self, points, method="linear", clip=True, **kwargs):
             values = self._interpolate(points[0])
             values = self.scale.inverse(values)
 
+        tiny = np.finfo(np.float32).tiny
```

Comment:
As this is a specific problem of the `LogScale`, I think we should move the code for the clipping there. So my proposal would be to define `self.is_tiny < (values < self.tiny)` in `LogScale._scale` and the re-apply the mask in `LogScale._inverse` with `values[self._is_tiny] = 0`. This will also get rid of the `try` and `except` block for quantities, because the interpolation scale is called with the units stripped-of.

Reply:
I moved the clipping code to LogScale.inverse, and I defined the mask only here as it need to be applied only to the output values. I added an other try and except block for the case where value is not an array.